include/utap/position.h

```diff
@@ -70,10 +71,12 @@ namespace UTAP
             uint32_t position;
             uint32_t offset;
             uint32_t line;
-            std::string path;
-            line_t(uint32_t position, uint32_t offset, uint32_t line, std::string path):
+            std::shared_ptr<std::string> path;
+            line_t(uint32_t position, uint32_t offset, uint32_t line, std::shared_ptr<std::string> path):
                 position(position), offset(offset), line(line), path{std::move(path)}
             {}
+
+            const std::string& xpath() const { return *path; }
```

Comment:
Assumes `assert(path != nullptr);`, perhaps the constructor is the best place for such an assertion.

Reply:
Great note, my assumption did not hold. `path` will sometimes be null